src/Compilers/CSharp/Portable/SymbolDisplay/SymbolDisplayVisitor.Members.cs

```diff
@@ -61,7 +61,8 @@ public override void VisitProperty(IPropertySymbol symbol)
             if (format.MemberOptions.IncludesOption(SymbolDisplayMemberOptions.IncludeType))
             {
                 var property = symbol as PropertySymbol;
-                if (property != null)
```

Comment:
Why do we have this cast to PropertySymbol.  That goes against a goal of SymbolDisplay as it means we can't build proper displays when passing in a VB symbol or an IDE synthesized symbol.  If it's because RefKind isn't on IPropertySymbol, then that needs ot change as the IDE (and other consumers) def needs to know about that and won't have access to the internal PropertySymbol.